pkg/httputils/httputils_test.go

```diff
@@ -0,0 +1,57 @@
+package httputils
+
+import (
+	"testing"
+)
+
+func TestDownload(t *testing.T) {
+	_, err := Download("http://docker.com")
+
+	if err != nil {
+		t.Errorf("Expected error to not exist when Download(http://docker.com)")
+	}
+
+	// Expected status code = 404
+	if _, err = Download("http://docker.com/abc1234567"); err == nil {
+		t.Errorf("Expected error to exist when Download(http://docker.com/abc1234567)")
+	}
+}
+
+func TestNewHTTPRequestError(t *testing.T) {
+	errorMessage := "Some error message"
+	httpResponse, _ := Download("http://docker.com")
+	if err := NewHTTPRequestError(errorMessage, httpResponse); err.Error() != errorMessage {
+		t.Errorf("Expected err to equal error Message")
+	}
+}
+
+func TestParseServerHeader(t *testing.T) {
+	serverHeader, err := ParseServerHeader("bad header")
+	if err.Error() != "Bad header: Failed regex match" {
+		t.Errorf("Should fail when header can not be parsed")
+	}
+
+	if serverHeader, err = ParseServerHeader("(bad header)"); err.Error() != "Bad header: '/' missing" {
+		t.Errorf("Should fail when header can not be parsed")
+	}
+
+	if serverHeader, err = ParseServerHeader("(without/spaces)"); err.Error() != "Bad header: Expected single space" {
+		t.Errorf("Should fail when header can not be parsed")
+	}
+
+	if serverHeader, err = ParseServerHeader("(header/with space)"); err != nil {
```

Comment:
If this passes then something is wrong.
The header should be something like "docker/1.8.0-dev (windows)" - notice there are no parens around the entire thing, just the OS.
I think the code in the parse func is wrong when it tries to split on "/" across the entire header instead of just the stuff before the first "(".
Can you take a stab at trying to fix the code and the then testcases in here?